ADR/0045-guidelines-for-developing-tasks.md

```diff
@@ -0,0 +1,54 @@
+# 45. Guidelines for developing tasks
+
+* Date: 2024-11-06
+
+## Status
+
+Proposed
+
+## Context
+
+One of the architectural goals for Konflux is extensibility. The core system provides opinionated [build pipelines]
+and [release pipelines], but lets users extend those and/or create their own. Beyond this goal, however, there has
+not been any clear guidance about where contributions should be added or the types of functionality that are typical
+for the stages of software development.
+
+## Decision
+
+* If any process is required to modify artifacts themselves, it should be done in [build pipelines]. This will ensure
+  that Konflux users can appropriately test and verify these artifacts before triggering releases with them.
+* Any tasks which need to be included in the artifacts' provenance (enabling verification with EC policies) must be
+  included in the [build pipelines].
+* Whenever possible, a task that just changes the format of an artifact (for example creating a tar from a container image)
+  should be performed in the same pipeline that produced the artifact. These modified artifacts should be attached using
+  the referrer's API and the blob digest reported via results to enable verification that the artifacts were produced
+  from a trusted task.
+* Artifact-modifying steps should generally be avoided in [release pipelines]. When interacting with produced artifacts,
+  the release pipeline should be limited to extracting artifacts from storage (i.e. an OCI registry), copying/pushing
+  artifacts to target locations, and updating metadata associated with the artifacts.
+* Reusable tasks or pipelines that are intended to be used in IntegrationTestScenarios should be contributed to
+  [pipeline samples].
+
```

Comment:
Do we want to add a bullet point to where/how Tasks are contributed to the build (or even release) pipelines here?